python/pyiceberg/catalog/__init__.py

```diff
@@ -97,7 +98,9 @@ def load_catalog(name: str, **properties: str | None) -> Catalog:
         if inferred_catalog_type := infer_catalog_type(conf):
             catalog_type = inferred_catalog_type
         else:
-            raise ValueError(f"Invalid configuration. Could not determine the catalog type: {properties}")
+            raise ValueError(
+                f"URI missing, please provide using --uri, the config or environment variable PYICEBERG_CATALOG__{name.upper()}__URI"
```

Comment:
I don't think this is necessarily true. Isn't the old error message more accurate?

Reply:
I've moved them to the `infer_catalog_type`, I think they make more sense there.